runtime/service.go

```diff
@@ -369,75 +303,125 @@ func (s *service) StartShim(ctx context.Context, id, containerdBinary, container
 
 	// If we're here, there is no pre-existing shim for this VMID, so we spawn a new one
 	defer func() {
-		log.G(ctx).WithField("id", id).Debug("closing shim socket")
+		log.G(ctx).WithField("id", containerID).Debug("closing shim socket")
 		shimSocket.Close()
 	}()
 
-	err = os.MkdirAll(s.vmDir(), 0700)
+	err = s.vmDir().Create()
```

Comment:
It depends on the circumstance IMO, but in this case I prefer what I have because it makes it easy to add stuff like
```
var err error
defer func() {
  if err != nil 
    cleanup()
  }
}()

err = doSomething()
if err != nil {
   return err
}
```

We aren't doing that at this exact moment in this function, but we actually will need to in the very near future when I fix #169 right after this PR is merged, so I'd prefer to leave it as is for now.

Reply:
```go
// This is fine too
if err = doSomething(); err != nil {
    return err
}
```

Reply:
Ohhh geeze, I've been confused about that for a while then. I always though that notation with `;` resulted in the setting of the variable `err` being scoped to the if block. I never thought to just do `=` instead of `:=` 😅

Makes sense though, will update.

Reply:
Haha, no worries. To be fair, the notation isn't very popular in languages.

Reply:
I started doing this, but then I encountered situations like this where I wanted to change
```
uuid, err := uuid.NewV4()
if err != nil {
  return "", err
}

s.vmID = uuid.String()
```
to
```
if uuid, err := uuid.NewV4(); err != nil {
  return "", err
}

s.vmID = uuid.String()
```

But that doesn't work because `uuid` is scoped to just the if conditional now. It gets even worse for the err checking with the `shimSocket`.

Honestly, given all that, I'd just prefer to keep it as is. It just seems simpler and less surprising to not use the `;` syntax unless there's a strong reason to.